docs/src/atomistic/overview.rst

```diff
@@ -0,0 +1,268 @@
+of Python, and PyTorch contains a compiler from Python to TorchScript code.
+After doing this translation, the model no longer depends on Python and can be
+executed directly inside simulation engines implemented in C, C++, Fortran, …
+This approach allow to define and tweak models as Python code, and then once
+they are working as intended, export them to a more stable representation to be
+used in simulations.
+
+In practice, models should be defined as custom :py:class:`torch.nn.Module`
+sub-class, following our :py:class:`ModelInterface`. New models can be written
+using this interface directly, and pre-existing models can use a small wrapper
+to convert from this interface to the model's existing input and output data.
+The models take as input a set of atomistic :py:class:`System` (typically a
+single one during simulations, and multiple systems during training); a set of
+``outputs`` requested by the engine, and should make prediction for all
+properties in the ``outputs``. All predictions are then returned to the engine
+in a dictionary of :py:class:`metatensor.torch.TensorMap`, one such tensor map
+for each property (i.e. energy, atomic charges, dipole, electronic density,
+chemical shielding, *etc.*)
+
+Once a model is defined and trained, it should be exported by constructing a
+:py:class:`MetatensorAtomisticModel`, and calling ``export`` on it. This class
+is a wrapper for the model that will handle unit conversions on input and
+outputs. It will also store metadata about the model (such as the authors, a
+list of references, …) and the model capabilities (what properties it can
+compute, which neighbors list the whole model requires, …). Optionally, this
+class can also check that both data provided by the engine and properties
+computed by the model follow the metatensor interface, which can be used to
+debug your code.
+
+Finally, the exported model can be loaded by simulation engines and used to run
+simulations and make predictions.
+
+
+.. _TorchScript: https://pytorch.org/docs/stable/jit.html
+
+Constrains on atomistic models
+------------------------------
+
+There are a couple of constrains on what a given model must do to be useable
+with metatensor, but appart from these you can do what you want inside the
+model!
+
+The main constrain is that the model must be compatible with `TorchScript`_,
+i.e. you must use either pure PyTorch code in the definition of your model, or
+implement a custom TorchScript extension for any operations where a pure PyTorch
+implementation is too slow or too much work.
+
+Your model should also only take data from our :py:class:`System` definition:
+atomic types and positions, simulation cell, and list of neighbors for different
+spherical cutoffs.
+
+If you need additional data that you can not compute inside the model (such as
+atomic spins, non-spherical particle orientation, *etc.*) you can use
+:py:meth:`System.get_data`, but this comes with significant caveats:
+
```

Comment:
Maybe we should say before we introduce the `data` attribute that people should talk to us if they want a quantity in the `systems`.

Maybe with a link to open a guthub issue.

Reply:
This "contact us" is just below, do you mean it should be moved above? This would feel a bit weird, since readers don't yet know what we are taking about. I can add a link to github though